_plugins/product-data-enricher.rb

```diff
@@ -0,0 +1,123 @@
+# This plugin enriches the product pages by setting or precomputing its metadata, so that it can be
+# easily consumed in layouts or plugins (such as the APIv1 plugin).
+module Jekyll
+  class ProductDataEnricher
+    class << self
+
+      TOPIC = "EndOfLife Product Data Enricher:"
+
+      def enrich(page)
+        Jekyll.logger.debug TOPIC, "Enriching #{page.name}"
+
+        set_id(page)
+        set_icon_url(page)
+        set_tags(page)
+
+        page.data["releases"].each { |release| enrich_release(page, release) }
+      end
+
+      def is_product?(page)
+        page.data['layout'] == 'product'
+      end
+
+      private
+
+      # Build the product id from the permalink.
+      def set_id(page)
+        page.data['id'] = page.data['permalink'][1..page.data['permalink'].length]
+      end
+
+      # Build the icon URL from the icon slug.
+      def set_icon_url(page)
+        if page['iconSlug']
+          page.data['iconUrl'] = "https://simpleicons.org/icons/#{page['iconSlug']}.svg"
+        end
+      end
+
+      # Explode tags space-separated string to a list if necessary.
+      # Also add the category as a default tag.
+      def set_tags(page)
+        tags = page.data['tags']
+
+        if tags
+          tags = (tags.kind_of?(Array) ? tags : tags.split)
+        else
+          tags = []
+        end
+
+        tags << page.data['category']
+        page.data['tags'] = tags
```

Comment:
Do we want to perhaps add the category as a tag here?

Reply:
This is done on the line before (48) ;)

Reply:
Oops, not sure how I missed that.